src/bin/dump-db.rs

```diff
@@ -0,0 +1,91 @@
+//! Helper to set up the database role and permissions for creating database
```

Comment:
I'm a bit confused by this binary. It seems like we're handling visibility when generating the dump anyway? Why do we need to create a new role and permissions, when it's programmatically generated from the same data used to determine what columns we access anyway?

Reply:
The visibility information in the configuration is for _columns_. There are a few tables we want to filter by _row_, though, and these are configured here.

I don't particularly like this design, but it does allow autogenerating the script, and testing whtether the column visibility information is up to date. If we are OK with ditching the unit test, we can go back to the slightly simpler desing in https://github.com/rust-lang/crates.io/issues/630#issuecomment-510380244.

Reply:
My issue is that we're using the same source both to generate these permissions as we are to generate the queries. If there's a bug in one there'll almost certainly be a bug in the other. I'm not sure that this is buying us any security for the complexity it gives.

Reply:
Using row security is not an _additional_ mechanism to filter the rows in this design. It's the _only_ mechanism to do so. The database role is not required to achieve some security gain, but to be able to use row security at all, since row security does not affect the owner of the tables.

So the reasoning is roughly
* We want to have a test to verify that the column information is up to date.
* Thus, we auto-generate the dump script from a column visibility configuration.
* Thus we need an independent mechanism to filter rows.
* Thus I came up with (ab)using row security to filter rows.

It may not be such a bright idea after all. From the review comments, it became clear to me that this design is far from obvious, and I also severely dislike having to create a new database role. Creating a role is a global change for the database server, instead of one that only affects the current database, so it can lead to interference between different databases on the same server, notably between the test and production databases in local installations.

I'll try to come up with a better TOML configuration format that also allows defining row filters, and that uses temporary views or subqueries to filter the rows.